doc/netplan.md

```diff
@@ -159,6 +159,13 @@ Virtual devices
     Note that **``rdnssd``**(8) is required to use RDNSS with networkd. No extra
     software is required for NetworkManager.
 
+``use-dns`` (bool)
+
+:   Enable DHCP to provide dns servers. On by default.
+
+    If you are planning to override your DHCP server provided DNS server,
+    set this to `no`.
+
```

Comment:
This description should note that this property is only used by networkd.

It also doesn't make it clear that (per your current schema design) it needs to be `dhcp: {use-dns: false}` - reading that I thought `use-dns` was at the same level as `dhcp4`/`dhcp6` etc.